### ISSUE TITLE
2d gaussian splatting support in splatfacto

### ISSUE THREAD

**Issue by infinitesnow** [nerfstudio-project/nerfstudio#3677]
**Is your feature request related to a problem? Please describe.**
I am trying to run 2dgs on a Polycam dataset. `nerfstudio` can parse such a dataset and outputs a `transform.json`.
`splatfacto` can load cameras from a `transform.json` file, but it seems to only support 3dgs.
`gsplat` has support for 2dgs, but doesn't seem to have a loader for `transform.json`, it wants a COLMAP dataset.  

**Describe the solution you'd like**
A switch to run splatfacto using 2d gaussian splatting.
 
**Describe alternatives you've considered**
Currently, I rerun the pose estimation in COLMAP, then run `gsplat` on it.

**Comment by peasant98** [comment]
Also wondering about this, i would down to help work on this too!

**Comment by ksh3456-a11y** [comment]
I am using a temporary patch on `Splatfacto.py`, modifying only the rasterization part.


```
def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:

       ..........

        # render, alpha, self.info = rasterization(
        #     means=means_crop,
        #     quats=quats_crop,  # rasterization does normalization internally
        #     scales=torch.exp(scales_crop),
        #     opacities=torch.sigmoid(opacities_crop).squeeze(-1),
        #     colors=colors_crop,
        #     viewmats=viewmat,  # [1, 4, 4]
        #     Ks=K,  # [1, 3, 3]
        #     width=W,
        #     height=H,
        #     packed=False,
        #     near_plane=0.01,
        #     far_plane=1e10,
        #     render_mode=render_mode,
        #     sh_degree=sh_degree_to_use,
        #     sparse_grad=False,
        #     absgrad=self.strategy.absgrad if isinstance(self.strategy, DefaultStrategy) else False,
        #     rasterize_mode=self.config.rasterize_mode,
        #     # set some threshold to disregrad small gaussians for faster rendering.
        #     # radius_clip=3.0,
        # )

        
        render, alpha, normals, normals_from_detph, distort, median, self.info, = rasterization_2dgs(
            means=means_crop,
            quats=quats_crop,  # rasterization does normalization internally
            scales=torch.exp(scales_crop),
            opacities=torch.sigmoid(opacities_crop).squeeze(-1),
            colors=colors_crop,
            viewmats=viewmat,  # [1, 4, 4]
            Ks=K,  # [1, 3, 3]
            width=W,
            height=H,
            near_plane=0.01,
            far_plane=1e10,
            sh_degree=sh_degree_to_use,
            render_mode=render_mode,
            sparse_grad=False,
            absgrad=self.strategy.absgrad if isinstance(self.strategy, DefaultStrategy) else False,
        )

        ..........
```

The camera poses are unchanged and are used exactly as in the original `Splatfacto` via `transforms.json`.

**Comment by peasant98** [comment]
Nice, I plan to add the exact formulation of 2D-GS, which differs in the Gaussians itself and a bit of the loss function.
